pkg/crypto/crypto.go

```diff
@@ -0,0 +1,77 @@
+/*
+	Package crypto provides functions to encode and decode data
+	Using aes crypto
+	Random flag defines random nonce sequence or same
+*/
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"io"
+)
+
+type cryptographer struct {
+	Key        []byte
+	RandomFlag bool
+}
+
+func NewCryptographer(key []byte) *cryptographer {
+	h := sha256.New()
+	key32 := make([]byte, 32)
+	copy(key32, h.Sum(key))
+	return &cryptographer{
+		Key:        key32,
+		RandomFlag: true,
+	}
+}
+
+func (c *cryptographer) Encode(value []byte) ([]byte, error) {
+	// Create a new Cipher Block from the key
+	// must be 16, 32, 64 bit key
+	block, err := aes.NewCipher(c.Key)
+	if err != nil {
+		return nil, fmt.Errorf("i cryptograper, encode method: invalid key: %w", err)
```

Comment:
What is the purpose if `i` prefix in all errors returned from `cryptographer`'s methods?